Supabase/src/commonMain/kotlin/io/github/jan/supabase/AccessTokenProvider.kt

```diff
@@ -0,0 +1,14 @@
+package io.github.jan.supabase
+
+/**
+ * Optional function for using a third-party authentication system with
+ * Supabase. The function should return an access token or ID token (JWT) by
+ * obtaining it from the third-party auth client library. Note that this
+ * function may be called concurrently and many times. Use memoization and
+ * locking techniques if this is not supported by the client libraries.
+ *
+ * When set, the Auth plugin from `auth-kt` cannot be used.
+ * Create another client if you wish to use Supabase Auth and third-party
+ * authentications concurrently in the same application.
+ */
+typealias AccessTokenProvider = () -> String
```

Comment:
This should be async, as some of the 3p providers returns the token asynchronously.